week_5/assignment/lottery-api/src/app.service.ts

```diff
@@ -1,8 +1,40 @@
 import { Injectable } from '@nestjs/common';
+import { ethers } from 'ethers';
+import * as tokenJson from './assets/MyToken.json'; // TODO
+import { Lottery } from '../lottery-contract/contracts/Lottery.sol'; // Not sure why the error
+import * as dotenv from "dotenv";
+import { StartLotteryDto } from './dtos/startLottery.dto';
+dotenv.config();
+
+// TODO: need a new deployed contract address
+const CONTRACT_ADDRESS = '' // old address '0x040f665db55e939e35e1abd684965f5184014aa9';
 
 @Injectable()
 export class AppService {
-  getHello(): string {
-    return 'Hello World!';
+  provider: ethers.providers.Provider;
+  contract: ethers.Contract;
+  lottery: Lottery;
+
+  constructor() {
+    this.provider = new ethers.providers.EtherscanProvider(
+      'goerli',
+      process.env.ETHERSCAN_API_KEY,
+    );
+
+    this.contract = new ethers.Contract(
+      CONTRACT_ADDRESS,
+      tokenJson.abi, // TODO
+      this.provider,
+    )
+  }
+
+  startLottery(body: StartLotteryDto) {
```

Comment:
this is where we have to attach it to the contract and then call the `openBets` function.